edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/cluster/KindClusterManager.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest.cluster
+{
+    using System;
+    using System.Threading.Tasks;
+    using CliWrap;
+    using k8s;
+
+    public class KindClusterManager : IKubernetesClusterManager
+    {
+        readonly string name;
+
+        public KindClusterManager(string name)
+        {
+            this.name = name;
+        }
+
+        public async Task Create()
+        {
+            await Cli.Wrap("/usr/local/bin/kind")
+                .SetArguments($@"create cluster --name ""{this.name}""")
+                .SetStandardOutputCallback(Console.WriteLine)
+                .SetStandardErrorCallback(Console.WriteLine)
+                .ExecuteAsync();
+        }
+
+        public async Task Delete()
+        {
+            await Cli.Wrap("/usr/local/bin/kind")
+                .SetArguments($"delete cluster --name {this.name}")
+                .ExecuteAsync();
+        }
+
+        public async Task<IKubernetes> GetClient()
+        {
+            string path = string.Empty;
+
+            await Cli.Wrap("/usr/local/bin/kind")
```

Comment:
Is it possible to create some input arguments kind path and kubeconfig file path?

Reply:
kind will create cluster and kubeconfig by itself. we can later request kubeconfig in order to work with kind cluster. I'm waiting for a good case where we have to add a parameter.